server/jetstream_api.go

```diff
@@ -2629,9 +2630,16 @@ func (s *Server) processStreamRestore(ci *ClientInfo, acc *Account, cfg *StreamC
 
 	var resp = JSApiStreamRestoreResponse{ApiResponse: ApiResponse{Type: JSApiStreamRestoreResponseType}}
 
-	// FIXME(dlc) - Need to close these up if we fail for some reason.
-	// TODO(dlc) - Might need to make configurable or stream direct to storage dir.
-	tfile, err := ioutil.TempFile("", "jetstream-restore-")
+	snapDir := path.Join(js.config.StoreDir, snapStagingDir)
+	if _, err := os.Stat(snapDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(snapDir, 0755); err != nil {
```

Comment:
`0750`, since its a new dir lets start strict rather than clean up later

Reply:
Should I change all of them? I know Phil wants that..

Reply:
ok made constants and changed all non-test versions.